torchtitan/components/mx.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# [Note] Getting the 'torchao' package:
+# This script requires the 'torchao' package to function correctly.
+# Please ensure you have this package installed from the appropriate repository.
+# You can obtain it from https://github.com/pytorch/ao by following the
+# installation instructions.
+
+# Note: Performance
+# Float8 experimental is intended to be ran under `torch.compile`` for competitive performance
+
+from importlib.metadata import version
+from importlib.util import find_spec
+from typing import Any, List
+
+import torch.nn as nn
+
+from torchtitan.config_manager import JobConfig, MX
+from torchtitan.distributed import ParallelDims
+from torchtitan.protocols.model_converter import (
+    ModelConverter,
+    register_model_converter,
+)
+from torchtitan.tools.logging import logger
+from torchtitan.tools.utils import has_cuda_capability
+
+# Maps titan recipe names to torchao mx recipe names
+NAME_MAP = {"mxfp8": "mxfp8_cublas"}
+
+
+class MXConverter(ModelConverter):
+    """Converts the linear layers of `model` to `MXLinear`."""
+
+    enabled: bool
+    filter_fqns: List[str] | str
+    mx_config: Any  # MXLinearConfig type when imported
+
+    def __init__(self, job_config: JobConfig, parallel_dims: ParallelDims):
+
+        self.enabled = False
+
+        if not has_cuda_capability(10, 0):
+            # Why no assert?
```

Comment:
unfinished business? Should remove or flip to assert.

Reply:
ohh I was pattern matching against Float8 which didn't assert and instead just pass through, I imagine for graceful hardware on older hardware but wanted to confirm